builtin/logical/consul/path_token.go

```diff
@@ -32,6 +47,9 @@ func pathToken(b *backend) *framework.Path {
 
 func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	role := d.Get("role").(string)
+	policies := d.Get("policies").([]string)
+	namespace := d.Get("consul_namespace").(string)
+	partition := d.Get("partition").(string)
```

Comment:
I can see why it feels tidier to group these statements together, but I would recommend declaring each right before it's used instead. Separating declaration and usage makes it more difficult to read and also increases the surface area for bugs, especially for mutable variables.

Reply:
I think you're right and that it's an easy choice for me to move them to where they are used in this case. What is your opinion of the same principle applied to a similar section in path_roles instead: https://github.com/hashicorp/vault/blob/1a9e1c8677dfb543ce606fad0fda442caff4d27f/builtin/logical/consul/path_roles.go#L149-L156
The first 5 were already present and I added just the last 2. Should I place the two I added with their code sections and leave the other five at the top? Or would dispersing the others to their appropriate sections of code start approaching on the "too much refactoring" territory in your opinion?

(At second glace, it seems only a couple of the others would move in the context of the function, but let's assume all 5 of the originals were to move for the sake of argument. I'm not sure which choice I'd prefer.)

Reply:
Refactored in https://github.com/hashicorp/vault/pull/13850/commits/dfec2cfed14973eb8d5cfe3c92f24bbc8e08c72f

I think it looks better with them moved after trying it out. Let me know what you think.

Reply:
On the set of 5, good question. It's in judgement call territory, so I wouldn't argue against either decision personally, but I like the way you've done it now.